ex02.exs

```diff
@@ -79,26 +78,25 @@ end
 ########################################
 
 defmodule Ex02 do
+  use Agent
 
   def new_counter(initial_value \\ 0) do
-    # « your code goes here»
+    {:ok, counter} = Agent.start_link(fn -> initial_value end)
+    counter
   end
 
   def next_value(agent) do
-    # « your code goes here»
+    value = Agent.get_and_update(agent, fn state -> {state, state + 1} end)
   end
 
   @global_name :my_global_agent
 
   def new_global_counter(initial_value \\ 0) do
-    # « your code goes here»
-    # You can give an agent a name by adding the
-    # `name: @global_name` option to start_link
+    Agent.start_link(fn -> initial_value end, name: @global_name)
+    
   end
 
   def global_next_value do
-    # « your code goes here»
-    # and you can use that name instead of a pid
-    # in calls such as `get_and_update`
+    value = Agent.get_and_update(:my_global_agent, fn state -> {state, state + 1} end)
```

Comment:
Why did you use `:my_global_agent` here? It kinda undermines the point of having the tag (because if someone changes thew value associatewd with the tag, it will change the name the process is registered under, but not  the name you use in get_and_update)

#-1